ui/ozone/platform/wayland/wayland_window.cc

```diff
@@ -149,20 +217,48 @@ PlatformImeController* WaylandWindow::GetPlatformImeController() {
 
 bool WaylandWindow::CanDispatchEvent(const PlatformEvent& native_event) {
   Event* event = static_cast<Event*>(native_event);
-  if (event->IsMouseEvent())
-    return has_pointer_focus_;
+  if (event->IsMouseEvent()) {
+    if (g_current_capture_ == this) {
+      if (has_pointer_focus_ ||
+          (child_window_->is_popup() && child_window_->has_focus())) {
+        return true;
+      } else {
+        return false;
+      }
+    } else {
+      return has_pointer_focus_;
+    }
+  }
   if (event->IsKeyEvent())
     return has_keyboard_focus_;
   return false;
 }
 
 uint32_t WaylandWindow::DispatchEvent(const PlatformEvent& native_event) {
+  Event* event = static_cast<Event*>(native_event);
+  if (event->IsLocatedEvent() && !has_pointer_focus_)
+    ConvertEventToDifferentWindow(event);
+
   DispatchEventFromNativeUiEvent(
       native_event, base::Bind(&PlatformWindowDelegate::DispatchEvent,
                                base::Unretained(delegate_)));
   return POST_DISPATCH_STOP_PROPAGATION;
 }
 
+void WaylandWindow::ConvertEventToDifferentWindow(ui::Event* event) {
```

Comment:
That's the same as in X11WindowOzone and DesktopWindowTreeHostX11. I'll name it the same way - ConvertEventLocationToCurrentWindowLocation